### PR TITLE
Fix auto exit

### PR DESCRIPTION
**Pull request by zhanghang1989**
*Issue #, if available:*

*Description of changes:*
safe start and auto exit for distributed search on multiple nodes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: autogluon/scheduler/remote/remote.py [hunk 1]
```diff
@@ -68,7 +68,10 @@ def __init__(self, remote_ip=None, port=None, local=False, ssh_username=None,
         else:
             remote_addr = (remote_ip + ':{}'.format(port))
             self.service = start_service(remote_ip, port)
-            super().__init__(remote_addr)
+            _set_global_remote_service(self.service)
+            import time
+            time.sleep(10)
+            super(Remote, self).__init__(remote_addr)
```
Comment (mseeger) [inline autogluon/scheduler/remote/remote.py]: Would
super().__init__(remote_addr)
also work here? I recall somebody changed all super calls to this simple format.

And is it OK to call the superclass __init__ in the middle of this method, and in an else branch?

**Comment by szha** [comment]
Job PR-386-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-386/1/index.html

File: autogluon/scheduler/remote/remote.py (hunk 1 above)
Comment (zhanghang1989) [reply autogluon/scheduler/remote/remote.py]: Agree. I have changed it to simple format.

I think it is okay to pass different args to __init__ method in an if else branch.

**Comment by zhanghang1989** [comment]
> Could you state what the issue is this is solving?
> This is beyond my expertise right now.

Thanks for reviewing the PR. 

Dask requires user to manually start worker at remote machines, which is not friendly to the user.
Therefore, we start a subprocess to execute terminal command at remote machines. Then we need to stop the subprocess after finish the program.

``_set_global_remote_service()`` would put the remote service into record.

Then during the program exit, these process will be automatically terminated:
https://github.com/awslabs/autogluon/blob/master/autogluon/scheduler/remote/remote.py#L161

**Comment by szha** [comment]
Job PR-386-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-386/2/index.html

**Comment by jwmueller** [comment]
Does this fix this error message that keeps popping up when Python is exited?

Task exception was never retrieved
future: <Task finished coro=<InProcConnector.connect() done, defined at /Users/jonasmue/virtual/tpot/lib/python3.7/site-packages/distributed/comm/inproc.py:285> exception=OSError("no endpoint for inproc address '10.119.109.44/74496/1'")>
Traceback (most recent call last):
  File "/Users/jonasmue/virtual/tpot/lib/python3.7/site-packages/distributed/comm/inproc.py", line 288, in connect
    raise IOError("no endpoint for inproc address %r" % (address,))
OSError: no endpoint for inproc address '10.119.109.44/74496/1'

**Comment by zhanghang1989** [comment]
> Does this fix this error message that keeps popping up when Python is exited?
> 
> Task exception was never retrieved
> future: <Task finished coro=<InProcConnector.connect() done, defined at /Users/jonasmue/virtual/tpot/lib/python3.7/site-packages/distributed/comm/inproc.py:285> exception=OSError("no endpoint for inproc address '10.119.109.44/74496/1'")>
> Traceback (most recent call last):
> File "/Users/jonasmue/virtual/tpot/lib/python3.7/site-packages/distributed/comm/inproc.py", line 288, in connect
> raise IOError("no endpoint for inproc address %r" % (address,))
> OSError: no endpoint for inproc address '10.119.109.44/74496/1'

No. This PR is only related to distributed training.

That error may be because of wrong dask/distributed version installed.

**Comment by jwmueller** [comment]
> That error may be because of wrong dask/distributed version installed.

Got it, it's annoying that it's difficult to uninstall the wrong versions of dask/distributed...
I repeatedly get messages from people who see this error and get worried.

**Comment by szha** [comment]
Job PR-386-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-386/3/index.html

**Comment by zhanghang1989** [comment]
The distributed training has been tested for newest DL AMI and the distributed training jobs are handled correctly. I am merging it now.

**Pull request merged**